sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -717,6 +721,18 @@ class CastChecks extends ExprChecks {
   val udtChecks: TypeSig = none
   val sparkUdtSig: TypeSig = STRING + UDT
 
+  protected def addDecimalSig(sig: TypeSig): TypeSig = {
+    val ansiModeValue = ansiMode.getOrElse(SparkSession.getActiveSession match {
+      case Some(session) => session.sessionState.conf.ansiEnabled
+      case None => false
+    })
+    if (ansiModeValue) {
+      sig + DECIMAL
+    } else {
+      sig
+    }
```

Comment:
nit: we can write this
```scala
    ansiMode.orElse(SparkSession.getActiveSession.map(_.sessionState.conf.ansiEnabled))
      .collect { case ansiModeValue if ansiModeValue => sig + DECIMAL }
      .getOrElse(sig)
```

Reply:
Thank you @gerashegalov! I've applied your suggestions. The functional programming style looks really cool!